doc/nrf/test_and_optimize.rst

```diff
@@ -1,17 +1,165 @@
 .. _test_and_optimize:
+.. _gs_testing:
+.. _serial_terminal_connect:
+.. _testing_rtt:
+.. _testing_rtt_connect:
+.. _putty:
+.. _testing_vscode:
 
 Testing and optimization
 ########################
 
 Each :ref:`applications` and :ref:`samples` in the |NCS| come with their own testing instructions.
-However, the |NCS| provides a common set of features and options for debugging and optimizing the application performance.
+Follow these instructions to make sure that the application runs as expected.
+
+Information about the current state of the application is usually provided through the LEDs or through :term:`Universal Asynchronous Receiver/Transmitter (UART)`, or through both.
+See the user interface section of the application's documentation for description of the LED states or available UART commands.
+
+To see the UART output, connect to the development kit (DK) with a terminal emulator.
+The default serial port settings are the following:
+
+.. list-table:: Default serial port connection settings for Nordic Semiconductor devices
+   :widths: auto
+   :header-rows: 1
+
+   * - Baud rate
+     - Data bits
+     - Stop bits
+     - Parity
+     - :term:`UART Hardware Flow Control (UART HWFC)`
+   * - 115200
+     - 8
+     - 1
+     - No parity
+     - None
+
+Use one of the following methods:
+
+.. tabs::
+
+   .. group-tab:: nRF Connect Serial Terminal
+
+      You can use the `nRF Connect Serial Terminal`_ app, which you can install from `nRF Connect for Desktop`_.
+      The nRF Connect Serial Terminal app establishes serial communication with the device, sends commands through UART, and also displays the UART output.
+
+      The application is available for Windows, Linux, and macOS.
```

Comment:
```suggestion
      The app is available for Windows, Linux, and macOS.
```
Since nothing new has been introduced, "application" in this context would be referring to the sample or application from the NCS, not Serial Terminal.